File: esports-backend/src/main/java/com/esports/entity/WalletTransaction.java
package com.esports.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Table(name = "wallet_transactions")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class WalletTransaction {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "wallet_id")
    private Wallet wallet;

    @ManyToOne
    @JoinColumn(name = "project_id")
    private Project project;

    @Column(name = "amount")
    private Double amount;

    @Enumerated(EnumType.STRING)
    @Column(name = "type")
    private TransactionType type;

    @Enumerated(EnumType.STRING)
    @Column(name = "status")
    private TransactionStatus status;

    @Column(name = "razorpay_payment_id")
    private String razorpayPaymentId;
    
    @Column(name = "bank_account")
    private String bankAccount;
    
    @Column(name = "ifsc_code")
    private String ifscCode;

    @Column(name = "created_at")
    private LocalDateTime createdAt = LocalDateTime.now();

    public enum TransactionType {
        ESCROW_HOLD, ESCROW_RELEASE, PAYMENT_RECEIVED, WITHDRAWAL, MILESTONE_PAYMENT
    }

    public enum TransactionStatus {
        PENDING, COMPLETED, FAILED
    }
}

File: Upkart1/Entities/WalletTransaction.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Upkart1.Entities.Enums;

namespace Upkart1.Entities
{
    /// <summary>
    /// Matches esports-backend entity: WalletTransaction (table: wallet_transactions).
    /// </summary>
    [Table("wallet_transactions")]
    public class WalletTransaction
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public long Id { get; set; }

        [Column("wallet_id")]
        public long WalletId { get; set; }
        public Wallet Wallet { get; set; } = null!;

        [Column("project_id")]
        public long? ProjectId { get; set; }
        public Project? Project { get; set; }

        [Column("amount")]
        public double Amount { get; set; }

        [Column("type")]
        public TransactionType Type { get; set; }

        [Column("status")]
        public TransactionStatus Status { get; set; }

        [Column("razorpay_payment_id")]
        public string? RazorpayPaymentId { get; set; }

        [Column("bank_account")]
        public string? BankAccount { get; set; }

        [Column("ifsc_code")]
        public string? IfscCode { get; set; }

        [Column("created_at")]
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }
}
